### ISSUE TITLE
Issues with authentication prompts when using profiles

### ISSUE THREAD

**Issue by traeok** [zowe/zowe-explorer-vscode#3329]
**Describe the bug**

_WIP_

There are a few issues with how we prompt for authentication when using profiles in v3.

- [ ] Extenders do not get credential prompts when using the APIs outside of Zowe Explorer actions
- [ ] We need to block before attempting to use a profile again with the invalid creds/token, to avoid continued attempts by the filesystem (some out of our control)
- [ ] We have to add error handling everywhere that the APIs are used to handle authentication errors
- [ ] While its not an issue, it would be nice to "lock" a profile so that we can wait / prompt for auth before making another request

**To Reproduce**

**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):**

- OS:
- Zowe Explorer Version:
- (Optional) Zowe CLI Version:
- (Optional) Are you using Secure Credential Store?

**Additional context**

**Comment by Buckminsterfullerene02** [comment]
> We need to block before attempting to use a profile again with the invalid creds/token, to avoid continued attempts by the filesystem (some out of our control) 

We have this issue big time, in that when our mainframe password is changed (every month), zowe consistently spams the mainframe with the incorrect credentials, causing mainframe lockout and having to go through the steps to unlock the account. It would be a great step forward to have this fixed on 3.x!

**Comment by davenice** [comment]
I have also hit the issue where it uses my creds, they fail, it prompts me to update, fails to update, and then tries them again.

Because there are 3 trees all using the same creds, it's easy to get locked out :-D

Would this issue be the right one to cover "don't reuse credentials after a 401, until they've changed"?

**Comment by traeok** [comment]
> Would this issue be the right one to cover "don't reuse credentials after a 401, until they've changed"?

Yes, I think this ties in with what @Buckminsterfullerene02 mentioned regarding the repeated attempts. This is primarily due to VS Code making requests for files in opened editors or virtual workspaces - multiple requests are made at the same time and our APIs do not currently support handling race conditions. I've been [working on a PoC](https://github.com/zowe/zowe-explorer-vscode/tree/refactor/handle-creds-in-api) for locking profiles to prevent repeated operations with the same invalid creds. Hoping to get this out in time for the 3.1.0 release.
